config/src/integration-test/java/org/springframework/security/config/annotation/authentication/ldap/LdapAuthenticationProviderConfigurerTests.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.annotation.authentication.ldap;
+
+import org.junit.Test;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LdapAuthenticationProviderConfigurerTests extends LdapBaseSpring {
+	@Test
+	public void authenticationmanagerSupportMultipleDefaultLdapContextsWithPortsDynamicallyAllocated() {
+		loadConfig(MultiLdapAuthenticationProvidersConfig.class);
+
+		getAuthenticationManager().authenticate(new UsernamePasswordAuthenticationToken("bob", "bobspassword"));
+	}
+
+	@Test
+	public void authenticationManagerSupportMultipleLdapContextWithDefaultRolePrefix() {
+		loadConfig(MultiLdapAuthenticationProvidersConfig.class);
+
+		Authentication authenticate = getAuthenticationManager().authenticate(new UsernamePasswordAuthenticationToken("bob", "bobspassword"));
+
+		assertThat(getAuthorities(authenticate)).contains(new SimpleGrantedAuthority("ROLE_DEVELOPERS"));
```

Comment:
We would like to move away from accessing the authorities using reflection and instead use `mockMvc`.
For an example of how to do this see commit 70db508.

Reply:
ok i'll check that this weekend

Reply:
Done (the LdapSprinClass is gone)

However, some comments about my last changes:
- there still are a lot of reflections (many assertions are about some private/protected properties)
- i had to change some @Configuration to @EnableWebSecurity to make mockMvc available